src/uuid.cr

```diff
@@ -321,6 +325,30 @@ struct UUID
     end
   {% end %}
 
+  # Generates an RFC9562-compatible v7 UUID, allowing the values to be sorted
+  # chronologically (with 1ms precision) by their raw or hexstring
+  # representation.
+  def self.v7(random r : Random = Random::Secure)
+    buffer = uninitialized UInt8[18]
+    value = buffer.to_slice
+
+    # Generate the first 48 bits of the UUID with the current timestamp. We
+    # allocated enough room for a 64-bit timestamp to accommodate the
+    # NetworkEndian.encode call here, but we only need 48 bits of it so we chop
+    # off the first 2 bytes.
+    IO::ByteFormat::NetworkEndian.encode Time.utc.to_unix_ms, value
+    value = value[2..]
```

Comment:
**suggestion** (nitpick): maybe extract a `timestamp` variable and rearrange a bit the above to make the steps more explicit?

```crystal
timestamp = Time.utc.to_unix_ms
buffer = uninitialized UInt8[18] # 2 bytes (skipped, see below) + 16 bytes (uuid)

# convert timestamp integer into bytes and skip the upper bytes (...)
IO::ByteFormat::NetworkEndian.encode timestamp, buffer.to_slice
value = buffer.to_slice[2..]
```